From 90891bf5fb8a5b82d7d0cc8e1195f2baaaa1a534 Mon Sep 17 00:00:00 2001
From: Gong Qianyu <Qianyu.Gong@freescale.com>
Date: Thu, 14 Apr 2016 14:01:28 +0800
Subject: [PATCH 06/74] common/board_f.c: change the macro name and remove it
 for PPC platforms

commit 18fb0e3cae64b03c51db7ba8b607e9031a844b99
[doesn't apply include/configs/*.h of powerpc]

For most PPC platforms, they will call the first get_clocks() in
init_sequence_f[] as they define CONFIG_PPC. CONFIG_SYS_FSL_CLK is
then defined to call the second get_clocks(), which should be
redundant for PPC.

Signed-off-by: Gong Qianyu <Qianyu.Gong@freescale.com>
Reviewed-by: York Sun <yorksun@freescale.com>
Integrated-by: Zhao Qiang <qiang.zhao@nxp.com>
---
 common/board_f.c             | 2 +-
 doc/README.fsl-clk           | 6 ++++++
 include/configs/mx7_common.h | 1 +
 3 files changed, 8 insertions(+), 1 deletion(-)
 create mode 100644 doc/README.fsl-clk

diff --git a/common/board_f.c b/common/board_f.c
index 613332e..b3a6693 100644
--- a/common/board_f.c
+++ b/common/board_f.c
@@ -806,7 +806,7 @@ static init_fnc_t init_sequence_f[] = {
 #if defined(CONFIG_BOARD_POSTCLK_INIT)
 	board_postclk_init,
 #endif
-#ifdef CONFIG_FSL_ESDHC
+#ifdef CONFIG_SYS_FSL_CLK
 	get_clocks,
 #endif
 #ifdef CONFIG_M68K
diff --git a/doc/README.fsl-clk b/doc/README.fsl-clk
new file mode 100644
index 0000000..9e83c24
--- /dev/null
+++ b/doc/README.fsl-clk
@@ -0,0 +1,6 @@
+Freescale system clock options
+
+	- CONFIG_SYS_FSL_CLK
+		Enable to call get_clocks() in board_init_f() for
+		non-PPC platforms and PCC 8xx platforms such as
+		TQM866M and TQM885D.
diff --git a/include/configs/mx7_common.h b/include/configs/mx7_common.h
index 04468b9..1b6edee 100644
--- a/include/configs/mx7_common.h
+++ b/include/configs/mx7_common.h
@@ -21,6 +21,7 @@
 #define CONFIG_MXC_GPT_HCLK
 #define CONFIG_SYSCOUNTER_TIMER
 #define CONFIG_SC_TIMER_CLK 8000000 /* 8Mhz */
+#define CONFIG_SYS_FSL_CLK
 
 /* Enable iomux-lpsr support */
 #define CONFIG_IOMUX_LPSR
-- 
2.1.0.27.g96db324

